solutions/devsprint-pedro-menezes-1/DeliveryAppChallenge/Modules/Product/Scenes/LeonardoAlmeida/Authentication/Login/LALoginViewController.swift

```diff
@@ -0,0 +1,240 @@
+    @IBOutlet weak var heightLabelError: NSLayoutConstraint!
+    
+    @IBOutlet weak var errorLabel: UILabel!
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    
+    @IBOutlet weak var loginButton: UIButton!
+    @IBOutlet weak var createAccountButton: UIButton!
+    @IBOutlet weak var showPasswordButton: UIButton!
+    
+    var passwordFieldIsSecure = true
+    var formLoginIsInvalid = false
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+       
+        let userIsLogged = verifyIfUserIsLogged()
+        if(userIsLogged) {
+            LALoginCoordinator.shared.navigateToHomeRoot()
+        }
+
+        #if DEBUG
+        emailTextField.text = "clean.code@devpass.com"
+        passwordTextField.text = "Abcde"
+        #endif
+
+        self.configView()
+        self.validateForm()
+    }
+    
+    open override var preferredStatusBarStyle: UIStatusBarStyle {
+        return .lightContent
+    }
+}
+
+// MARK: - Configurações de Layout
+extension LALoginViewController {
+    
+    func configView() {
+        heightLabelError.constant = 0
+        
+        loginButton.layer.cornerRadius = loginButton.frame.height / 2
+        loginButton.backgroundColor = .blue
+        loginButton.setTitleColor(.white, for: .normal)
+        loginButton.isEnabled = true
+
+        showPasswordButton.tintColor = .lightGray
+
+        createAccountButton.layer.cornerRadius = createAccountButton.frame.height / 2
+        createAccountButton.layer.borderWidth = 1
+        createAccountButton.layer.borderColor = UIColor.blue.cgColor
+        createAccountButton.setTitleColor(.blue, for: .normal)
+        createAccountButton.backgroundColor = .white
+        
+        emailTextField.setDefaultColor()
+        passwordTextField.setDefaultColor()
+        
+        let gesture = UITapGestureRecognizer(target: self, action: #selector(didClickView))
+        view.addGestureRecognizer(gesture)
+        view.isUserInteractionEnabled = true
+    }
+
+    @objc
+    func didClickView() {
+        view.endEditing(true)
+    }
+    
+    func configPasswordTextField() {
+        let imageIcon: UIImage? = passwordFieldIsSecure ? .init(systemName: "eye") : .init(systemName: "eye.slash")
+        passwordTextField.isSecureTextEntry = passwordFieldIsSecure
+        showPasswordButton.setImage((imageIcon ?? UIImage()).withRenderingMode(.alwaysTemplate), for: .normal)
+    }
+    
+
+}
+
+// MARK: - Edição de Campos
+extension LALoginViewController {
+    @IBAction func emailBeginEditing(_ sender: Any) {
+        if formLoginIsInvalid {
+            resetErrorLogin(emailTextField)
+        } else {
+            emailTextField.setEditingColor()
+        }
+    }
+    
+    @IBAction func passwordBeginEditing(_ sender: Any) {
+        if formLoginIsInvalid {
+            resetErrorLogin(passwordTextField)
+        } else {
+            passwordTextField.setEditingColor()
+        }
```

Comment:
essa funcao e a de cima estão muito parecidas
quando for assim, pense se tem como reaproveitar codigo 
no caso, vc pode criar uma funçao que faz isso e vc passa o textfield de parâmetro